src/pages/iou/request/step/IOURequestStepCategory.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import CategoryPicker from '@components/CategoryPicker';
+import Text from '@components/Text';
+import transactionPropTypes from '@components/transactionPropTypes';
+import useLocalize from '@hooks/useLocalize';
+import compose from '@libs/compose';
+import Navigation from '@libs/Navigation/Navigation';
+import reportPropTypes from '@pages/reportPropTypes';
+import useThemeStyles from '@styles/useThemeStyles';
+import * as IOU from '@userActions/IOU';
+import ROUTES from '@src/ROUTES';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import StepScreenWrapper from './StepScreenWrapper';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /* Onyx Props */
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    transaction: transactionPropTypes,
+
+    /** The report attached to the transaction */
+    report: reportPropTypes,
+};
+
+const defaultProps = {
+    report: {},
+    transaction: {},
+};
+
+function IOURequestStepCategory({
+    report,
+    route: {
+        params: {transactionID, backTo},
+    },
+    transaction,
+}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const navigateBack = () => {
+        Navigation.goBack(backTo || ROUTES.HOME);
+    };
+
+    /**
+     * @param {Object} category
+     * @param {String} category.searchText
+     */
+    const updateCategory = (category) => {
+        IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, category.searchText);
+        navigateBack();
+    };
```

Comment:
Here, we have forgot to handle the case of de-selecting a category, which caused https://github.com/Expensify/App/issues/32872